### PR TITLE
Update cdap-java-spark-archetype

### PR DESCRIPTION
**Pull request by rohitsinha54**
This is just a copy and paste from our SparkKMeansExample. 
- Bumped up the spark core version to cdap dependency version (which is currently 1.3.1)
- Tested creation of project from archetype
- Tested project build and unit tests
- Tested being able to open in IDE

### CODE REVIEW

File: cdap-archetypes/cdap-spark-java-archetype/src/main/resources/archetype-resources/src/main/java/SparkPageRankApp.java [hunk 1]
```diff
@@ -93,15 +127,27 @@ public SparkSpecification configure() {
...
+    public static final String RANKS_SERVICE_PATH = "rank";
+
     @UseDataSet("ranks")
     private ObjectStore<Integer> ranks;
 
-    @Path("rank")
-    @GET
-    public void getRank(HttpServiceRequest request, HttpServiceResponder responder, @QueryParam("url") String url) {
+    @Path(RANKS_SERVICE_PATH)
+    @POST
```
Comment (sreevatsanraman) [inline cdap-archetypes/cdap-spark-java-archetype/src/main/resources/archetype-resources/src/main/java/SparkPageRankApp.java]: why is getRank a POST call?

Comment (rohitsinha54) [inline cdap-archetypes/cdap-spark-java-archetype/src/main/resources/archetype-resources/src/main/java/SparkPageRankApp.java]: We need to use POST call as the query string being sent has / and : which can't be sent through path params and the UI does not support query parameters. 
I renamed the method and removed get from it for some clarity.

**Comment by sreevatsanraman** [comment]
LGTM :+1:

**Comment by rohitsinha54** [comment]
Doing one final testing to create from archetype before merge.

**Pull request merged**